ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/fillers/CoreProblemFiller.java

```diff
@@ -59,33 +59,31 @@ private void fillCnec(Cnec cnec) {
     private void fillRangeAction(RangeAction rangeAction) {
         double minValue = rangeAction.getMinValue(linearRaoData.getNetwork());
         double maxValue = rangeAction.getMaxValue(linearRaoData.getNetwork());
-        rangeAction.getApplicableRangeActions().forEach(applicableRangeAction ->
-            applicableRangeAction.getNetworkElements().forEach(networkElement -> {
-                Identifiable pNetworkElement = linearRaoData.getNetwork().getIdentifiable(networkElement.getId());
-                if (pNetworkElement instanceof TwoWindingsTransformer) {
-                    TwoWindingsTransformer transformer = (TwoWindingsTransformer) pNetworkElement;
-                    double currentAlpha = transformer.getPhaseTapChanger().getCurrentStep().getAlpha();
-                    double minAlpha = transformer.getPhaseTapChanger().getStep((int) minValue).getAlpha();
-                    double maxAlpha = transformer.getPhaseTapChanger().getStep((int) maxValue).getAlpha();
-                    if (currentAlpha >= minAlpha && currentAlpha <= maxAlpha) {
-                        linearRaoProblem.addRangeActionVariable(
+        rangeAction.getNetworkElements().forEach(networkElement -> {
+            Identifiable pNetworkElement = linearRaoData.getNetwork().getIdentifiable(networkElement.getId());
+            if (pNetworkElement instanceof TwoWindingsTransformer) {
```

Comment:
you mean
` if (rangeAction.getType().equals(PST_RANGE_ACTION))`

@TheMaskedTurtle :heart: